providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/UnleashProvider.java

```diff
@@ -0,0 +1,171 @@
+package dev.openfeature.contrib.providers.unleash;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderEventDetails;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import io.getunleash.DefaultUnleash;
+import io.getunleash.Unleash;
+import io.getunleash.UnleashContext;
+import io.getunleash.Variant;
+import io.getunleash.util.UnleashConfig;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.getunleash.Variant.DISABLED_VARIANT;
+
+/**
+ * Provider implementation for Unleash.
+ */
+@Slf4j
+public class UnleashProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "Unleash Provider";
+    public static final String NOT_IMPLEMENTED =
+        "Not implemented - provider does not support this type. Only boolean is supported.";
```

Comment:
This is not true. Unleash support `boolean`, `string`, `json` and `csv`. We also expect to add `number` as variant payload soon.

Reply:
Can you please clarify on boolean ? in Unleash payload, I don't see an option for boolean type, see screenshot.

What do you think about the following concepts, and if you have any suggestion for this ?
- Boolean evaluation gets feature enabled status.
- String evaluation gets feature variant value.

<img width="267" alt="image" src="https://github.com/open-feature/java-sdk-contrib/assets/77168114/51c0cbae-8745-4e62-8000-de6049534bd6">

Reply:
I agree with your comment for "boolean value". 

In Unleash a toggle can either be enabled or disabled. In the case of being enabled it can also take a variant. The variant may take values in the form of payload, which can take different types. 

If it is disabled the user can specify the fallback variant to be used, which can also take a payload.

Reply:
I think this is one of the main points in this provider. We can separate to boolean/string cases per my interpertation of the [spec](getStringEvaluation(String key, String defaultValue, EvaluationContext ctx)):

- _getStringEvaluation(String key, String defaultValue, EvaluationContext ctx)_
The method expects defaultValue, not the value of default variant, this is not exactly by what you described. How I see it in the implementation here, is to fit the generic case of evaluation by default value of this method.
- getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx)
Since there is no explicit "boolean" payload type, I thought to return the "enabled" status of the flag here, and not the payload. This will act like "feature is enabled" flag.

What do you think of the concepts here ?
CC. @toddbaert

Reply:
This is going to be the main challenge of this provider (though we do have a minority of other providers that have the same challenge and have done something similar). It's mostly a documentation issue. I think what you've done is the most reasonable choice, given that Unleash supports string/json/csv payloads in addition to a boolean enabled/disabled state.

The other option could be use use [flagMetadata](https://openfeature.dev/specification/types#flag-metadata) to carry the enabled property, but I think that would be a poor user experience.

You could also use the constructor of your provider to allow some customization/fallback of this behavior, at the cost of some complexity.